gapic/src/main/com/google/gapid/views/GotoCommand.java

```diff
@@ -92,8 +92,12 @@ protected Control createCustomArea(Composite parent) {
     protected void buttonPressed(int buttonId) {
       String[] strings = text.getText().split("\\.");
       value = new ArrayList<Long>();
-      for (String s : strings) {
-        value.add(Long.parseLong(s));
+      try {
+        for (String s : strings) {
+          value.add(Long.parseLong(s));
+        }
+      } catch ( NumberFormatException e) {
+        // TODO(apbodnar) Show some sort of warning/error
```

Comment:
What's the right thing to do here?

Reply:
There are two approaches we can take:
1) proactively parse the input whenever it changes, and disable the OK button and show a message if the input is invalid.
2) parse only when the user clicks OK, and don't close the dialog and show a message if the input is invalid. If you don't call `super.buttonPressed(..)`, the dialog will not close and stay around.

Personally, I prefer option 1.

Reply:
Option 1 is much nicer :)

Reply:
Updated to disable the OK button and show a simple error message.  Though there's a good chance i'm doing something suboptimal with SWT